src/test/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps/GenerateReportsStepTest.groovy

```diff
@@ -0,0 +1,187 @@
+            step.additionalSettings*.value == resultValues
+        where:
+            scenario                            |given                  | resultNames               | resultValues
+            "remove empty additional settings"  | additionalSettings    | ["setting1","setting2"]   | ["value1","value2"]
+            "handle empty list"                 | []                    | []                        | []
+            "handle null"                       | null                  | []                        | []
+
+    }
+
+    def "setReportIds should '#scenario'"() {
+        given:
+            def step = new UploadReportsStep("http://localhost:8085", "credId123")
+
+        when:
+            step.setReportIds(given)
+
+        then:
+            step.reportIds == result
+        where:
+            scenario            |given                          | result
+            "remove empty ids"  | ["1", "", "2", "  ", "3"]     | ["1", "2", "3"]
+            "handle empty list" | []                            | []
+            "handle null"       | null                          | []
+    }
+
+    @Unroll
+    def "Should handle report generation for generator '#generator'"() {
+        given:
+            def logger = Mock(PrintStream)
+            def step = new GenerateReportsStep(generator)
+            step.setReportIds(["1", "2"])
+            def execution = new GenerateReportsStep.Execution(step, stepContext)
+            GroovyMock(RestApiClientFactory, global: true)
+
+        and:
+            envVars.expand(generator) >> generator
+            envVars.expand("1") >> "1"
+            envVars.expand("2") >> "2"
+            listener.logger >> logger
+            RestApiClientFactory.getRestApiClient(*_) >> apiClient
+
+            apiClient.generateReport(_, _) >> new GenerationResult("Success", message, "folder")
+            channel.call(_) >> { MasterToSlaveCallable callable ->
+                return callable.call()
+            }
+
+        when:
+            def results = execution.run()
+
+        then:
+            results.size() == 2
+            results.every {
+                        it.generationResult == "Success" &&
+                        it.generationMessage == message &&
+                        it.reportOutputDir == "folder"
+            }
+            1 * logger.println("Generating ${generator} reports...")
+            1 * logger.println("- Generating ${generator} report format for report id 1...")
+            1 * logger.println("- Generating ${generator} report format for report id 2...")
+            2 * logger.println("  -> Success${messagePrint}")
+            1 * logger.println("${generator} reports generated successfully.")
+
+
+        where:
+            generator   | message       | messagePrint
+            'HTML'      | "message"     | " (message)"
+            'JSON'      | ""            | ""
+    }
+
+    def "Should call getAllReportIds if '#scenario'"() {
```

Comment:
We should think about the test scenario and what happen if some values are set.
Such like if report ids are set or not. the `apiClient.getAllReportIds()` should be called or not and the logging should be done or not